web/html/src/manager/maintenance/maintenance-windows-edit.js

```diff
@@ -0,0 +1,287 @@
+/* eslint-disable */
+'use strict';
+
+const React = require("react");
+
+const ReactDOM = require("react-dom");
+const AsyncButton = require("components/buttons").AsyncButton;
+const Button = require("components/buttons").Button;
+const ModalButton = require("components/dialog/ModalButton").ModalButton;
+const DangerDialog = require("components/dialog/DangerDialog").DangerDialog;
+const { Combobox } = require("components/combobox");
+const { InnerPanel } = require("components/panels/InnerPanel");
+const { Form } = require('components/input/Form');
+const { Text } = require('components/input/Text');
+const { Radio } = require('components/input/Radio');
+const { Check } = require('components/input/Check');
+
+class MaintenanceWindowsEdit extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            type: props.type,
+            calendarDataText: "",
+            selectedCalendar: 0
+        };
+
+        if(this.isEdit()) {
+            this.setSchedule(this.props.schedule);
+        }
+    }
+
+    setSchedule = (schedule) => {
+        Object.assign(this.state, schedule);
```

Comment:
`this.state` should be treated as *immutable*. Depending on the data shape maybe you can use something like `this.setState({schedule})`?